src/test/FixedPointMathLib.t.sol

```diff
@@ -6,12 +6,54 @@ import {DSTestPlus} from "./utils/DSTestPlus.sol";
 import {FixedPointMathLib} from "../utils/FixedPointMathLib.sol";
 
 contract FixedPointMathLibTest is DSTestPlus {
-    function testExp() public {
-        assertEq(FixedPointMathLib.expWad(-1e18), 367879441171442321);
+    function testExpWadDown() public {
+        assertEq(FixedPointMathLib.expWadDown(-42139678854452767551), 0);
+
+        assertEq(FixedPointMathLib.expWadDown(-3e18), 49787068367863942);
+        assertEq(FixedPointMathLib.expWadDown(-2e18), 135335283236612691);
+        assertEq(FixedPointMathLib.expWadDown(-1e18), 367879441171442321);
+
+        assertEq(FixedPointMathLib.expWadDown(-0.5e18), 606530659712633423);
+        assertEq(FixedPointMathLib.expWadDown(-0.3e18), 740818220681717866);
+
+        assertEq(
+            FixedPointMathLib.expWadDown(0),
+            999999999999999999
+            // True value: 1000000000000000000 (off by 1 wei)
```

Comment:
This actually annoys me quite a bit.

Reply:
lmk adding to the constant term works without impacting the other vars

another way thats only a couple extra opcodes would be doing 

```
           assembly {
            	r := add(r, iszero(x))
            }
```

right at the end